### PR TITLE
Add new Timeline layout

### PR DESCRIPTION
**Pull request by Nitwel**

### CODE REVIEW

File: app/src/layouts/timeline/index.ts [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+				loading.value = true;
+
+				const result = await api.get(getEndpoint(collection.value!), {
+					params: {
+						groupBy: [`year(${dateField.value})`, `month(${dateField.value})`, `day(${dateField.value})`],
+						aggregate: {
+							count: '*',
+						},
+						limit: -1,
+						// sort: sort.value,
```
Comment (azrikahar) [inline app/src/layouts/timeline/index.ts:160]: WindowsTerminal_l9reRopU5W

enabling sort here seems to trigger the similar error in #10050, where `order by "tasks"."end_date" desc` in turn necessitates `end_date` to be added to groupBy. Adding it does resolve it, but the grouping by end_date ended up messing up the group instead. Not sure what's happening here.

**Comment by danieljwestman** [comment]
@Nitwel Anything going on here? I think I've seen a branch for the timeline layout a few days ago...

**Comment by Nitwel** [comment]
@azrikahar started reviewing this a few days ago and adding the final touches. But I don't know when this will be merged.

**Comment by sksar** [comment]
Any updates on this one? @rijkvanzanten @Nitwel

**Comment by azrikahar** [comment]
@sksar updates will be included here, if there's any 👍 There's also the status shown on the sidebar for "Engineering and More" project board, so hopefully that should help give a bit more clue on the current status as well (which is `Wishlist` at the moment).

**Comment by rijkvanzanten** [comment]
Heya, thanks for your PR! Our small team has been attempting to work through a backlog of external contributions for the better part of a year, but in that time many have become too stale to merge or the change is no longer desired/relevant.. 
We've updated the [contributing guidelines](https://github.com/directus/directus/blob/main/contributing.md#pull-requests) and are working on [making more resources available](https://github.com/directus/directus/discussions/17977) to make sure that doesn’t happen again 😬 In the meantime, we'll have to close this PR for now..

**Pull request closed without merging**